clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -432,6 +432,24 @@ public ConfigDef withClientSaslSupport() {
     }
 
     /**
+     * Filter configs that are not defined in this configuration definition.
+     *
+     * @param props The configs to be filtered.
+     * @return Filtered configs.
+     */
+    public Map<String, Object> filter(Map<String, Object> props) {
```

Comment:
That's a good point. Note that we only need the config key names and that is safe to expose.